tests/test_benchmarks.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2022 Graphcore Ltd. All rights reserved.
+
+import tempfile
+import os
+import pathlib
+import pytest
+from examples_utils.benchmarks.environment_utils import get_git_commit_hash
+
+cwd = pathlib.Path.cwd()
+
+@pytest.fixture()
+def teardown():
+    yield
+    os.chdir(cwd)
+
+def test_git_commit_hash_in_repo():
+    os.chdir(pathlib.Path(__file__).parent)
+    hash = get_git_commit_hash()
+    assert(is_sha_1(hash))
+
+
+def test_git_commit_hash_out_of_repo():
+    os.chdir(tempfile.gettempdir())
+    not_a_hash = get_git_commit_hash()
+    assert "Not a git repo" in not_a_hash
```

Comment:
Good cover of the directory edge case. I think you need `@pytest.fixture(autouse=True)` otherwise the fixture is not used.

In pytest you also have the `monkeypatch` fixture which builds that functionality:

```suggestion

def test_git_commit_hash_in_repo(monkeypatch):
    monkeypatch.chdir(pathlib.Path(__file__).parent)
    hash = get_git_commit_hash()
    assert(is_sha_1(hash))


def test_git_commit_hash_out_of_repo(monkeypatch, tmp_path):
    monkeypatch.chdir(tmp_path)
    not_a_hash = get_git_commit_hash()
    assert "Not a git repo" in not_a_hash
```

The `tmp_path` fixture gives you a temporary directory which gets torn down on succesful end of the test.